crates/core/src/live_socket/channel.rs

```diff
@@ -143,24 +167,58 @@ impl LiveChannel {
         // TODO: This should probably take the event closure to send changes back to swift/kotlin
         let document = self.document.clone();
         let events = self.channel.events();
+        let statuses = self.channel.statuses();
         loop {
-            let event = events.event().await?;
-            match event.event {
-                Event::Phoenix { phoenix } => {
-                    error!("Phoenix Event for {phoenix:?} is unimplemented");
-                }
-                Event::User { user } => {
-                    if user == "diff" {
-                        let Payload::JSONPayload { json } = event.payload else {
-                            error!("Diff was not json!");
-                            continue;
+            let event = events.event().fuse();
+            let status = statuses.status().fuse();
+
+            pin_mut!(event, status);
+
+            select! {
+               e = event => {
+                   let e = e?;
+                   match e.event {
```

Comment:
Not that it matters that much but I think you might be able to get away with `match e?.event`.